sdk/core/core/src/az_http_policy.c

```diff
@@ -92,8 +96,35 @@ AZ_NODISCARD az_result az_http_pipeline_policy_logging(
     az_http_response * const response) {
   (void)data;
 
-  // Authentication logic
-  return az_http_pipeline_nextpolicy(p_policies, hrb, response);
+  if (data == NULL) {
+    return az_http_pipeline_nextpolicy(p_policies, hrb, response);
+  }
+
+  clock_t const threshold
+      = (data == NULL) ? 0 : ((az_log_options const *)data)->slow_response_threshold;
+  clock_t const start = clock();
+
+  az_result const result = az_http_pipeline_nextpolicy(p_policies, hrb, response);
+
+  clock_t const end = clock();
+
+  if (data != NULL) {
+    clock_t const duration = end - start;
+    if (duration >= threshold && az_log_should_write(AZ_LOG_SLOW_RESPONSE)) {
+      uint8_t log_msg_buf[200];
+      az_span_builder log_msg_bldr
+          = az_span_builder_create((az_mut_span)AZ_SPAN_FROM_ARRAY(log_msg_buf));
+
+      AZ_RETURN_IF_FAILED(az_span_builder_append_unsigned_number(
```

Comment:
Failure to log MUST not fail the operation. If the buffer is not large enough to hold the log entry, just log what you have and keep going.

Reply:
Done.